crates/orderbook/src/database/orders.rs

```diff
@@ -140,7 +170,12 @@ async fn insert_order(order: &Order, ex: &mut PgConnection) -> Result<(), Insert
             } else {
                 InsertionError::DbError(err)
             }
-        })
+        })?;
+    database::orders::insert_interactions(ex, &order.uid, &interactions)
+        .await
```

Comment:
You probably want to insert the order and interactions in the same database transaction.